doc/sonic-multi-architecture/sonic_arm_support.md

```diff
@@ -0,0 +1,199 @@
+
+To do such cross compilation, we can make use of binfmt-misc to run target arch binary using qemu-static binary to run on the host cpu architecture. 
+
+```sh
+sonic-slave-arm64
+sonic-slave-armhf
+```
+
+qemu static binaries need to be installed and docker for multiarch/qemu-user-static:register is enabled to run.
+
+### Miscellaneous
+
+Architecture specific packages need to installed or ignored.
+Like ixgbe and grub are specific to X86 architecture, which need to be excluded.
+
+
+### Platform
+
+Same platform or board can have variants in CPU vendor. To address this, platform can be made ARCH specific, and customized changes can be added in this platform specific make infra.
+
+```sh
+platform/marvell-armhf/docker-syncd-mrvl-rpc.mk
+platform/marvell-armhf/docker-syncd-mrvl-rpc/99-syncd.conf
+platform/marvell-armhf/docker-syncd-mrvl-rpc/Dockerfile.j2
+platform/marvell-armhf/docker-syncd-mrvl-rpc/ptf_nn_agent.conf
+platform/marvell-armhf/docker-syncd-mrvl.mk
+platform/marvell-armhf/docker-syncd-mrvl/Dockerfile.j2
+platform/marvell-armhf/docker-syncd-mrvl/start.sh
+platform/marvell-armhf/docker-syncd-mrvl/supervisord.conf
+platform/marvell-armhf/docker-syncd-mrvl/syncd.sh
+platform/marvell-armhf/libsaithrift-dev.mk
+platform/marvell-armhf/linux-kernel-armhf.mk
+platform/marvell-armhf/one-image.mk
+platform/marvell-armhf/platform.conf
+platform/marvell-armhf/rules.mk
+platform/marvell-armhf/sai.mk
+platform/marvell-armhf/sai/Makefile
+```
+
+#### Rule/makefile
+
+Hardcoded "amd64" need to be replaced with Makefile variable which hold the target architecture.
+* amd64
+* armhf
+* arm64
+
+```sh
+rules/bash.mk
+rules/docker-base-stretch.mk
+rules/docker-base.mk
+rules/docker-ptf.mk
+rules/docker-snmp-sv2.mk
+rules/frr.mk
+rules/gobgp.mk
+rules/hiredis.mk
+rules/iproute2.mk
+rules/isc-dhcp.mk
+rules/libnl3.mk
+rules/libteam.mk
+rules/libyang.mk
+rules/linux-kernel.mk
+rules/lldpd.mk
+rules/lm-sensors.mk
+rules/mpdecimal.mk
+rules/python3.mk
+rules/quagga.mk
+rules/radvd.mk
+rules/redis.mk
+rules/sairedis.mk
+rules/smartmontools.mk
+rules/snmpd.mk
+rules/socat.mk
+rules/swig.mk
+rules/swss-common.mk
+rules/swss.mk
+rules/tacacs.mk
+rules/telemetry.mk
+rules/thrift.mk
+slave.mk
+src/bash/Makefile
+src/hiredis/Makefile
+src/iproute2/Makefile
+src/isc-dhcp/Makefile
+src/libnl3/Makefile
+src/libteam/Makefile
+src/lm-sensors/Makefile
+src/mpdecimal/Makefile
+src/python3/Makefile
+src/radvd/Makefile
+src/redis/Makefile
+src/smartmontools/Makefile
+src/snmpd/Makefile
+src/socat/Makefile
+src/tacacs/nss/Makefile
+src/tacacs/pam/Makefile
+src/thrift/Makefile
+
+```
+
+### Repo list
+Below repo sources list need to updated as the azure debian repo doesn't have arm packages 
+
+
+```sh
+files/apt/sources.list-armhf
+files/build_templates/sonic_debian_extension.j2
+
+```
+
+#### Onie Image
+
+Onie image configuration and build script should be updated for the uboot specific environment for ARM.
```

Comment:
thanks for the update. can we get more details on this one? how sonic installer works on arm?

Reply:
Done